src/cargo/util/config.rs

```diff
@@ -424,11 +424,25 @@ impl ConfigValue {
 }
 
 fn homedir() -> Option<PathBuf> {
-    let cargo_home = env::var_os("CARGO_HOME").map(PathBuf::from);
+    let cargo_home = get_cargo_home_dir();
     let user_home = env::home_dir().map(|p| p.join(".cargo"));
     return cargo_home.or(user_home);
 }
 
+fn get_cargo_home_dir() -> Option<PathBuf> {
+    let cargo_home = env::var_os("CARGO_HOME").map(PathBuf::from);
+    match cargo_home {
+        Some(path) => if !path.is_absolute() {
+            let mut absolute_path = env::current_dir().unwrap();
```

Comment:
Nah `cwd` will already be a `&Path` and then `join` is defined for anything that can go into an `OsStr`, where `home` is already an `OsString`